apix/v1/inferencepool_types.go

```diff
@@ -0,0 +1,283 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// InferencePool is the Schema for the InferencePools API.
+//
+// +kubebuilder:object:root=true
+// TODO: change the annotation once it gets officially approved
+// +kubebuilder:metadata:annotations="api-approved.kubernetes.io=unapproved, experimental-only"
+// +kubebuilder:subresource:status
+// +kubebuilder:storageversion
+// +genclient
+type InferencePool struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// +required
+	Spec InferencePoolSpec `json:"spec,omitzero"`
+
+	// Status defines the observed state of InferencePool.
+	//
+	// +kubebuilder:default={parent: {{parentRef: {kind: "Status", name: "default"}, conditions: {{type: "Accepted", status: "Unknown", reason: "Pending", message: "Waiting for controller", lastTransitionTime: "1970-01-01T00:00:00Z"}}}}}
+	// +optional
+	Status InferencePoolStatus `json:"status,omitzero"`
+}
+
+// InferencePoolList contains a list of InferencePool.
+//
+// +kubebuilder:object:root=true
+type InferencePoolList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []InferencePool `json:"items"`
+}
+
+// InferencePoolSpec defines the desired state of InferencePool
+type InferencePoolSpec struct {
+	// Selector determines which Pods are members of this inference pool.
+	// It matches Pods by their labels only within the same namespace; cross-namespace
+	// selection is not supported.
+	//
+	// The structure of this LabelSelector is intentionally simple to be compatible
+	// with Kubernetes Service selectors, as some implementations may translate
+	// this configuration into a Service resource.
+	//
+	// +required
+	Selector LabelSelector `json:"selector,omitempty,omitzero"`
+
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=1
+	// +listType=map
+	// +listMapKey=number
+	// +required
+	TargetPorts []Port `json:"targetPorts,omitempty"`
+
+	// Extension configures an endpoint picker as an extension service.
+	// +required
+	ExtensionRef Extension `json:"extensionRef,omitempty,omitzero"`
+}
+
+type Port struct {
+	// Number defines the port number to access the selected model server Pods.
+	// The number must be in the range 1 to 65535.
+	//
+	// +required
+	Number PortNumber `json:"number,omitempty"`
+}
+
+// Extension specifies how to configure an extension that runs the endpoint picker.
+type Extension struct {
+	// Group is the group of the referent.
+	// The default value is "", representing the Core API group.
+	//
+	// +optional
+	// +kubebuilder:default=""
+	Group *Group `json:"group,omitempty"`
```

Comment:
what types in the core API group would be valid to point to for this? Is defaulting to "" going to bite people who forget about API group and think just their kind is sufficient to specify some non-core type? Required with no default to require specifying the group and kind together seems more likely to avoid accidents

Reply:
We expect >95% of references to be to Service long term, so we wanted that experience to be as simple as possible. Currently Service is the only kind supported by Gateway implementations, but leaving this here allows implementations to support custom kinds as needed. 

This is copying what we did [in Gateway API](https://github.com/kubernetes-sigs/gateway-api/blob/3f2b1eee89144073a245404b6055c2d6461715dd/apis/v1/object_reference_types.go#L119-L125) for backendRefs in Routes. All Gateways support references to Service, and that represents >90% of use. Some implementations have expanded that to support ServiceImport, and now InferencePool. So far I haven't seen or heard of any issues where a user set kind but failed to set group.

Reply:
The crd showed it clearly to users that  `The default value is "", representing the Core API group.` So there won't be any misunderstanding here. Resolve it for now.